swiftide-integrations/src/anthropic/chat_completion.rs

```diff
@@ -58,7 +58,7 @@ impl ChatCompletion for Anthropic {
 
         let request = anthropic_request
             .build()
-            .map_err(|e| ChatCompletionError::LLM(Box::new(e)))?;
+            .map_err(|e| LanguageModelError::PermanentError(e.into()))?;
```

Comment:
Small nit, these lines feel more verbose than they should, doesn't `.map_err(LanguageModelError::PermanentError)` work?

Reply:
oh shit yeah that would be a lot better